scripts/west_commands/zspdx/walker.py

```diff
@@ -194,60 +244,52 @@ def setupDocuments(self):
-
-        # also set up app sources package
-        cfgPackageApp = PackageConfig()
-        cfgPackageApp.name = "app-sources"
-        cfgPackageApp.spdxID = "SPDXRef-app-sources"
-        # relativeBaseDir is app sources dir
-        cfgPackageApp.relativeBaseDir = self.cm.paths_source
-        pkgApp = Package(cfgPackageApp, self.docApp)
-        self.docApp.pkgs[pkgApp.cfg.spdxID] = pkgApp
+    def setupSDKDocument(self):
+        # set up SDK document
+        cfgSDK = DocumentConfig()
+        cfgSDK.name = "sdk"
+        cfgSDK.namespace = self.cfg.namespacePrefix + "/sdk"
+        cfgSDK.docRefID = "DocumentRef-sdk"
+        self.docSDK = Document(cfgSDK)
+
+        # also set up zephyr sdk package
+        cfgPackageSDK = PackageConfig()
+        cfgPackageSDK.name = "sdk"
+        cfgPackageSDK.spdxID = "SPDXRef-sdk"
+        # relativeBaseDir is SDK dir
+        cfgPackageSDK.relativeBaseDir = self.sdkPath
+        pkgSDK = Package(cfgPackageSDK, self.docSDK)
+        self.docSDK.pkgs[pkgSDK.cfg.spdxID] = pkgSDK
 
         # create DESCRIBES relationship data
         rd = RelationshipData()
         rd.ownerType = RelationshipDataElementType.DOCUMENT
-        rd.ownerDocument = self.docApp
+        rd.ownerDocument = self.docSDK
         rd.otherType = RelationshipDataElementType.PACKAGEID
-        rd.otherPackageID = cfgPackageApp.spdxID
+        rd.otherPackageID = cfgPackageSDK.spdxID
         rd.rlnType = "DESCRIBES"
 
         # add it to pending relationships queue
         self.pendingRelationships.append(rd)
 
-        if self.cfg.includeSDK:
-            # set up SDK document
-            cfgSDK = DocumentConfig()
-            cfgSDK.name = "sdk"
-            cfgSDK.namespace = self.cfg.namespacePrefix + "/sdk"
-            cfgSDK.docRefID = "DocumentRef-sdk"
-            self.docSDK = Document(cfgSDK)
-
-            # also set up zephyr sdk package
-            cfgPackageSDK = PackageConfig()
-            cfgPackageSDK.name = "sdk"
-            cfgPackageSDK.spdxID = "SPDXRef-sdk"
-            # relativeBaseDir is SDK dir
-            cfgPackageSDK.relativeBaseDir = self.sdkPath
-            pkgSDK = Package(cfgPackageSDK, self.docSDK)
-            self.docSDK.pkgs[pkgSDK.cfg.spdxID] = pkgSDK
-
-            # create DESCRIBES relationship data
-            rd = RelationshipData()
-            rd.ownerType = RelationshipDataElementType.DOCUMENT
-            rd.ownerDocument = self.docSDK
-            rd.otherType = RelationshipDataElementType.PACKAGEID
-            rd.otherPackageID = cfgPackageSDK.spdxID
-            rd.rlnType = "DESCRIBES"
+    # set up Documents before beginning
+    def setupDocuments(self):
+        log.dbg("setting up placeholder documents")
 
-            # add it to pending relationships queue
-            self.pendingRelationships.append(rd)
+        self.setupBuildDocument()
+
+        try:
+            with open(os.path.join(self.cfg.buildDir, "zephyr_meta.yml")) as file:
```

Comment:
Instead of hardcoded `zephyr_meta.yml` file here, then such values should idealy be passed to the script as argument.
I do notice that `walker.py` is called through the west extension command, which make such possibility a bit harder.

It could be considered if the CMakeCache should be used to keep the meta file name in sync.